index.js

```diff
@@ -54,12 +53,66 @@ class OktaStatusError extends Error {
   }
 }
 
+class ConfigurationValidationError extends Error {}
+const findDomainURL = 'https://bit.ly/finding-okta-domain';
+const findAppCredentialsURL = 'https://bit.ly/finding-okta-app-credentials';
+const copyCredentialsMessage = 'You can copy it from the Okta Developer Console ' +
+  'in the details for the Application you created. ' +
+  `Follow these instructions to find it: ${findAppCredentialsURL}`;
+
+const isHttps = new RegExp('^https://');
+const hasDomainAdmin = /-admin.(okta|oktapreview|okta-emea).com/;
+
+function assertIssuer(issuer, testing = {}){
+  const copyMessage = 'You can copy your domain from the Okta Developer ' +
+    'Console. Follow these instructions to find it: ' + findDomainURL;
+
+  if (testing.disableHttpsCheck) {
+    const httpsWarning = 'Warning: HTTPS check is disabled. ' +
+      'This allows for insecure configurations and is NOT recommended for production use.';
+    /* eslint-disable-next-line no-console */
+    console.warn(httpsWarning);
+  }
+
+  if (!issuer) {
+    throw new ConfigurationValidationError('Your Okta URL is missing. ' + copyMessage);
+  } else if (!testing.disableHttpsCheck && !issuer.match(isHttps)) {
+    throw new ConfigurationValidationError(
+      'Your Okta URL must start with https. ' +
+      `Current value: ${issuer}. ${copyMessage}`
+    );
+  } else if (issuer.match(/{yourOktaDomain}/)) {
+    throw new ConfigurationValidationError('Replace {yourOktaDomain} with your Okta domain. ' + copyMessage);
+  } else if (issuer.match(hasDomainAdmin)) {
+    throw new ConfigurationValidationError(
+      'Your Okta domain should not contain -admin. ' +
+      `Current value: ${issuer}. ${copyMessage}`
+    );
+  }
+}
+
+function assertClientId(clientId){
+  if (!clientId) {
+    throw new ConfigurationValidationError('Your client ID is missing. ' + copyCredentialsMessage);
+  } else if (clientId.match(/{clientId}/)) {
+    throw new ConfigurationValidationError('Replace {clientId} with the client ID of your Application. ' + copyCredentialsMessage);
+  }
+}
+
+function assertRedirectUri(redirectUri){
+  if (!redirectUri) {
+    throw new ConfigurationValidationError('Your redirect URI is missing.');
+  } else if (redirectUri.match(/{redirectUri}/)) {
+    throw new ConfigurationValidationError('Replace {redirectUri} with the redirect URI of your Application.');
+  }
+}
+
```

Comment:
copy and convert javascript based [configuration-validation](https://github.com/okta/okta-oidc-js/blob/master/packages/configuration-validation/src/lib.ts)